pipeline_tools/create_envelope.py

```diff
@@ -4,79 +4,94 @@
+    response.raise_for_status()
     envelope_url = get_entity_url(response.json(), 'submissionEnvelopes')
+    return envelope_url
 
-    # 2. Create envelope, get analysis and submission urls
+
+@retry(reraise=True, wait=wait_exponential(multiplier=1, max=RETRY_SECONDS), stop=stop_after_delay(TIMEOUT_SECONDS))
+def create_submission_envelope(envelope_url, auth_headers):
     print('Creating submission envelope at {0}'.format(envelope_url))
     response = requests.post(envelope_url, '{}', headers=auth_headers)
-    check_status(response.status_code, response.text)
+    response.raise_for_status()
     envelope_js = response.json()
-    analyses_url = get_entity_url(envelope_js, 'processes')
-    print('Creating analysis at {0}'.format(analyses_url))
-    submission_url = get_entity_url(envelope_js, 'submissionEnvelope')
-    with open('submission_url.txt', 'w') as f:
-        f.write(submission_url)
+    return envelope_js
 
-    # 3. Create analysis, get input bundles url, file refs url
-    with open(analysis_json_path) as f:
-        analysis_json_contents = json.load(f)
+
+@retry(reraise=True, wait=wait_exponential(multiplier=1, max=RETRY_SECONDS), stop=stop_after_delay(TIMEOUT_SECONDS))
+def create_analysis(analyses_url, auth_headers, analysis_json_contents):
+    print('Creating analysis at {0}'.format(analyses_url))
     response = requests.post(analyses_url, headers=auth_headers, data=json.dumps(analysis_json_contents))
-    check_status(response.status_code, response.text)
+    response.raise_for_status()
     analysis_js = response.json()
-    input_bundles_url = get_entity_url(analysis_js, 'add-input-bundles')
-    file_refs_url = get_entity_url(analysis_js, 'add-file-reference')
+    return analysis_js
 
-    # 4. Add input bundles
+
+@retry(reraise=True, wait=wait_exponential(multiplier=1, max=RETRY_SECONDS), stop=stop_after_delay(TIMEOUT_SECONDS))
+def add_input_bundles(input_bundles_url, auth_headers, analysis_json_contents):
     print('Adding input bundles at {0}'.format(input_bundles_url))
     input_bundle_uuid = get_input_bundle_uuid(analysis_json_contents)
     bundle_refs_js = json.dumps({"bundleUuids": [input_bundle_uuid]}, indent=2)
     print(bundle_refs_js)
     response = requests.put(input_bundles_url, headers=auth_headers, data=bundle_refs_js)
-    check_status(response.status_code, response.text)
+    response.raise_for_status()
 
-    # 5. Add file references
-    print('Adding file references at {0}'.format(file_refs_url))
-    output_files = get_output_files(analysis_json_contents, schema_version)
-    for file_ref in output_files:
-        print('Adding file: {}'.format(file_ref['fileName']))
-        response = requests.put(file_refs_url, headers=auth_headers, data=json.dumps(file_ref))
-        check_status(response.status_code, response.text)
-
-
-def check_status(status, response_text, expected='2xx'):
-    """Check that status is in range 200-299 or the specified range, if given.
```

Comment:
I'm not sure if `raise_for_status` can fully replace this function? It seems this function is not only checking if the status code is 2xx, but also compare the status code with the `expected` parameter. 

The question is do we really have a use case that requires us inputting the `expected` parameter? If not, we can replace this with `raise_for_status`!

Reply:
Looks good to me! Thanks for this change!